### PR TITLE
Add back the otlpmetrichttp exporter

### PR DESCRIPTION
**Pull request by MrAlias**
Part of #2807

Notable differences from the `main` branch exporter:

- `NewUnstarted` is removed. The `otlpmetric.Client` no longer has a concept of started or unstarted, the new exporter returned is always assumed to be ready to export.
- `NewClient` is removed. Use `otlpmetricgrpc.New` to create a usable `metric.Exporter`.
- The client does not handle concurrency requirements, that is a concern of `otlpmetric.Exporter`.
- The client does not handle shutdown once requirements, that is a concern of the `otlpmetric.Exporter`.
- The client internals no longer hold both configurations as fields. A request is built during creation that contains on the configurable options and is used as a clone for all uploads made.
- The deprecated `WithMaxAttempts` and `WithBackoff` options are dropped.

This PR does not include full integration testing. I plan to address that in a follow-on PR that is being tracked in https://github.com/open-telemetry/opentelemetry-go/issues/2807.

### CODE REVIEW

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+		default:
+		}
+
+		request.reset(iCtx)
+		resp, err := c.httpClient.Do(request.Request)
+		if err != nil {
+			return err
+		}
+
+		var rErr error
```
Comment (jmacd) [inline exporters/otlp/otlpmetric/otlpmetrichttp/client.go:166]: There's discussion in #3106, in which I would recommend restructuring the way errors are handled in this block of code. I would be glad to follow-up with the same feedback here _after this PR_.

**Review by jmacd — APPROVED** [review]

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go (hunk 1 above)
Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:166]: > There's discussion in #3106, in which I would recommend restructuring the way errors are handled in this block of code. I would be glad to follow-up with the same feedback here _after this PR_.

Absolutely, that sounds like a good plan to me :+1:

File: exporters/otlp/otlpmetric/otlpmetrichttp/config.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+// environment variable is set, and this option is not passed, the path
+// contained in that variable value will be used. If both are set,
+// OTEL_EXPORTER_OTLP_METRICS_ENDPOINT will take precedence.
+//
+// By default, if an environment variable is not set, and this option is not
+// passed, "/v1/metrics" will be used.
+func WithURLPath(urlPath string) Option {
+	return wrappedOption{oconf.WithURLPath(urlPath)}
+}
+
```
Comment (MadVikingGod) [inline exporters/otlp/otlpmetric/otlpmetrichttp/config.go:108]: 👍 For removing the deprecated `WithMaxAttempts` and `WithBackoff`

**Review by MadVikingGod — APPROVED** [review]

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/oconf"
+	"go.opentelemetry.io/otel/sdk/metric"
+	colmetricpb "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	metricpb "go.opentelemetry.io/proto/otlp/metrics/v1"
+)
+
+// New returns an OpenTelemetry metric Exporter. The Exporter can be used with
+// a PeriodicReader to export OpenTelemetry metric data to an OTLP receiving
+// endpoint using protobufs over HTTP.
+func New(_ context.Context, opts ...Option) (metric.Exporter, error) {
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:45]: Yeah, good question. I went both ways on this.

By including it we ...

- Ensure this is equivalent with the metric gRPC exporter [`New`](https://github.com/open-telemetry/opentelemetry-go/blob/f20ead1de4842d35ea9b7b37089aca152adc4f21/exporters/otlp/otlpmetric/otlpmetricgrpc/client.go#L38-L52), the trace HTTP [`New`](https://github.com/open-telemetry/opentelemetry-go/blob/13ddf7d31e389c891689eca29428bcc72ef98572/exporters/otlp/otlptrace/otlptracehttp/exporter.go#L23-L26), and the trace gRPC [`New`](https://github.com/open-telemetry/opentelemetry-go/blob/13ddf7d31e389c891689eca29428bcc72ef98572/exporters/otlp/otlptrace/otlptracegrpc/exporter.go#L23-L26)
- Allow us to use a context in the future if we discover a need to

However, by including it, we are asking users to provide us something that is not used.

I'm open to opinions here.

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+		transport := ourTransport.Clone()
+		transport.TLSClientConfig = cfg.Metrics.TLSCfg
+		httpClient.Transport = transport
+	}
+
+	format := "https://%s%s"
+	if cfg.Metrics.Insecure {
+		format = "http://%s%s"
+	}
+	rawURL := fmt.Sprintf(format, cfg.Metrics.Endpoint, cfg.Metrics.URLPath)
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:96]: Not a strong one. The `http.NewRequest` accepts a rawURL as a string and then uses the the `url.Parse` function to validate the string, returning an error for invalid values. I figured generate a string directly here and skipping the import, but it is not a strong reason to do so. I'll revert to what I had before.

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmathieu) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:46]: I think passing the context for consistency (and possible future use) makes sense.

**Review by dashpole — APPROVED** [review]

**Pull request merged**